### ISSUE TITLE
Should VS/vsUTCS be installed system-wide? Per user? Either?

### ISSUE THREAD

**Issue by stephengtuggy** [vegastrike/Assets-Masters#27]
Hmm. I see your point.

I guess this raises another question: Should Vega Strike / vsUTCS be a system-wide installation? A per-user installation? Should they support both configurations?

_Originally posted by @stephengtuggy in https://github.com/vegastrike/Assets-Production/issues/49#issuecomment-776344084_

**Comment by BenjamenMeyer** [comment]
For Mac we package a DMG which users can decide to install to `~/Applications` for a user-level install instead of `/Applications` for a system-wide install.

For Linux we advocate using the package managers which only do a system-wide install. Users can of course build on their own for a user-specific install or even an alternate installation path.

Windows... generally its good practice for Windows installers to offer the user the option to install to the system wide location (`%ProgramFiles%`) or to their user profile. I'm not sure how hard it is to do that with our Windows installer right now which if I am not mistaken is simply generated via CPack. We probably should set [CPACK_NSIS_MODIFY_PATH]( https://cmake.org/cmake/help/latest/cpack_gen/nsis.html#variable:CPACK_NSIS_MODIFY_PATH) to `ON` instead of `OFF`; that would probably also require some changes in the scripting to make some things more dynamic but would probably provide a better user experience

**Comment by BenjamenMeyer** [comment]
we need to add the ability to the Assets Installer to be able to select a VS Engine.

@stephengtuggy @kheckwrecker does NSIS record the VS Engine installations to the Windows Registry?

We may need to figure out a way to integrate a custom NSIS script into the Assets installer to support this.

**Comment by stephengtuggy** [comment]
> we need to add the ability to the Assets Installer to be able to select a VS Engine.
> 
> [@stephengtuggy](https://github.com/stephengtuggy) [@kheckwrecker](https://github.com/kheckwrecker) does NSIS record the VS Engine installations to the Windows Registry?
> 
> We may need to figure out a way to integrate a custom NSIS script into the Assets installer to support this.

Yes, I think it does, same as any Windows installer records information on what's installed. Finding where that information is recorded might be the tricky part.

**Comment by BenjamenMeyer** [comment]
> > we need to add the ability to the Assets Installer to be able to select a VS Engine.
> > [@stephengtuggy](https://github.com/stephengtuggy) [@kheckwrecker](https://github.com/kheckwrecker) does NSIS record the VS Engine installations to the Windows Registry?
> > We may need to figure out a way to integrate a custom NSIS script into the Assets installer to support this.
> 
> Yes, I think it does, same as any Windows installer records information on what's installed. Finding where that information is recorded might be the tricky part.

If we can figure out how to do that, then I'm all for enabling this. If we cannot, then we should close this as a `won't do`.

Ideally, on Windows the Assets installer would auto-detect the installed VS Engines, select the one it was released with by default but enable the user to select another. We might want to add a flag on the VS Engine for it to report the API version via an exit code to support this too (assuming it wasn't also recorded into the Windows Registry with the installer information or otherwise easily available via NSIS) so it could determine which are compatible and which are not via a simple query.

Not sure how complex this all is. I'm okay leaving this as a proposal until the details can be figured out.
